src/main/java/bgu/spl/mics/application/MI6Runner.java

```diff
@@ -1,11 +1,131 @@
+import bgu.spl.mics.application.passiveObjects.MI6RunnerInfo.AgentInfo;
+import bgu.spl.mics.application.passiveObjects.MI6RunnerInfo.MI6RunnerInfo;
+import bgu.spl.mics.application.publishers.TimeService;
+import bgu.spl.mics.application.subscribers.Intelligence;
+import bgu.spl.mics.application.subscribers.M;
+import bgu.spl.mics.application.subscribers.Moneypenny;
+import bgu.spl.mics.application.subscribers.Q;
+import com.google.gson.Gson;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This is the Main class of the application. You should parse the input file, create the different instances of the
+ * objects, and run the system. In the end, you should output serialized objects.
  */
 public class MI6Runner {
+    private static final int ARGS_NUM = 3;
+    private static final int INPUT_FILE = 0;
+    private static final int INVENTORY_FILE = 1;
+    private static final int DIARY_FILE = 2;
+
     public static void main(String[] args) {
-        // TODO Implement this
+        if (args.length != ARGS_NUM) {
+            System.err.println("Invalid num of arguments");
+            return;
+        }
+        FileReader json;
+        try {
+            json = new FileReader(args[INPUT_FILE]);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            return;
+        }
+        Gson gson = new Gson();
+        MI6RunnerInfo info = gson.fromJson(json, MI6RunnerInfo.class);
+        Inventory.getInstance().load(info.getInventory().toArray(new String[]{}));
+        List<Agent> tempSquad = new ArrayList<>();
+        for (AgentInfo agentInfo : info.getSquad()) {
+            tempSquad.add(new Agent(agentInfo.getSerialNumber(), agentInfo.getName()));
+        }
+        Squad.getInstance().load(tempSquad.toArray(new Agent[]{}));
+        int subscribersNum = info.getServices().getMoneypenny() + info.getServices().getM() +
+                info.getServices().getIntelligence().size() + 1; // + 1 for Q
+        CountDownLatch latch = new CountDownLatch(subscribersNum);
+        ExecutorService e = Executors.newFixedThreadPool(subscribersNum + 1); // + 1 for Time service.
+        e.execute(new Q(latch));
+        for (int i = 0; i < info.getServices().getMoneypenny(); i++) {
+            e.execute(new Moneypenny(i, latch));
+        }
+        for (int i = 0; i < info.getServices().getM(); i++) {
+            e.execute(new M(i, latch));
+        }
+        for (int i = 0; i < info.getServices().getIntelligence().size(); i++) {
+            e.execute(new Intelligence(i, latch, info.getServices().getIntelligence().get(i).getMissions()));
+        }
+        TimeService timeService = new TimeService(info.getServices().getTime(), latch);
+        e.execute(timeService);
```

Comment:
This could also probably get a function, call it createServices maybe and have it return the executor